### PR TITLE
Improved Stamina API

### PR DESCRIPTION
**Pull request by FUTURE-SL**

### CODE REVIEW

**Review by louis1706 — APPROVED** [review]

File: Exiled.API/Features/Player.cs [hunk 1]
```diff
@@ -953,6 +955,16 @@ public float Stamina
...
+        /// <summary>
+        /// Gets or sets the stamina usage multiplier.
+        /// </summary>
+        public float StaminaUsageMultiplier { get; set; } = 1f;
+
+        /// <summary>
+        /// Gets or sets the stamina regen multiplier.
+        /// </summary>
+        public float StaminaRegenMultiplier { get; set; } = 1f;
+
```
Comment (louis1706) [inline Exiled.API/Features/Player.cs:967]: Before someone else merge this things

 should this be inside Player::Role.As<FpcRole>() ?

Comment (FUTURE-SL) [reply Exiled.API/Features/Player.cs:967]: Mb no, To example, IsUsingStamina not in Role class, so StaminaUsageMultiplier and StaminaRegenMultiplier should be at Player class too

Comment (louis1706) [reply Exiled.API/Features/Player.cs:967]: that why i ask actually the good part of making it inside Role it's than it's would be reset everytime player change Role

Comment (FUTURE-SL) [reply Exiled.API/Features/Player.cs]: I made some changes

**Review by louis1706 — APPROVED** [review]

File: Exiled.API/Features/Roles/FpcRole.cs [hunk 2]
```diff
@@ -126,6 +128,30 @@ public bool MovementDetected
         /// </summary>
         public bool IsInvisible { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether or not the player should use stamina system for current live.
```
Comment (Thundermaker300) [inline Exiled.API/Features/Roles/FpcRole.cs:132]:
```suggestion
        /// Gets or sets a value indicating whether or not the player should use stamina system. Resets on death.
```

File: Exiled.API/Features/Roles/FpcRole.cs [hunk 3]
```diff
@@ -126,6 +128,30 @@ public bool MovementDetected
...
+            set
+            {
+                if (!value)
+                    Owner.ResetStamina();
+                isUsingStamina = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the stamina usage multiplier for current live.
```
Comment (Thundermaker300) [inline Exiled.API/Features/Roles/FpcRole.cs:146]:
```suggestion
        /// Gets or sets the stamina usage multiplier. Resets on death.
```

File: Exiled.API/Features/Roles/FpcRole.cs [hunk 4]
```diff
@@ -126,6 +128,30 @@ public bool MovementDetected
...
+        /// </summary>
+        public float StaminaUsageMultiplier { get; set; } = 1f;
+
+        /// <summary>
+        /// Gets or sets the stamina regen multiplier for current live.
```
Comment (Thundermaker300) [inline Exiled.API/Features/Roles/FpcRole.cs:151]:
```suggestion
        /// Gets or sets the stamina regen multiplier. Resets on death.
```

**Review by Thundermaker300 — CHANGES_REQUESTED** [review]

File: Exiled.API/Features/Player.cs [hunk 5]
```diff
@@ -1067,9 +1093,17 @@ public bool BadgeHidden
         public bool IsInPocketDimension => CurrentRoom?.Type is RoomType.Pocket;
 
         /// <summary>
-        /// Gets or sets a value indicating whether or not the player should use stamina system.
+        /// Gets or sets a value indicating whether or not the player should use stamina system for current live.
```
Comment (Thundermaker300) [inline Exiled.API/Features/Player.cs:1096]:
```suggestion
        /// Gets or sets a value indicating whether or not the player should use stamina system. Resets on death.
```

File: Exiled.API/Features/Player.cs [hunk 6]
```diff
@@ -954,6 +954,32 @@ public float Stamina
...
+            get => Role is FpcRole fpcRole ? fpcRole.StaminaUsageMultiplier : 1f;
+            set
+            {
+                if (Role is FpcRole fpcRole)
+                    fpcRole.StaminaUsageMultiplier = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the stamina regen multiplier for current live.
```
Comment (Thundermaker300) [inline Exiled.API/Features/Player.cs:971]:
```suggestion
        /// Gets or sets the stamina regen multiplier. Resets on death.
```

File: Exiled.API/Features/Player.cs [hunk 7]
```diff
@@ -954,6 +954,32 @@ public float Stamina
             set => StaminaStat.CurValue = value;
         }
 
+        /// <summary>
+        /// Gets or sets the stamina usage multiplier for current live.
```
Comment (Thundermaker300) [inline Exiled.API/Features/Player.cs:958]:
```suggestion
        /// Gets or sets the stamina usage multiplier. Resets on death.
```

**Review by Thundermaker300 — APPROVED** [review]

**Comment by Thundermaker300** [comment]
Tested?

**Comment by FUTURE-SL** [comment]
> Tested?

Yes

File: Exiled.API/Features/Player.cs [hunk 8]
```diff
@@ -1994,7 +2028,17 @@ public int RemoveItem(Func<Item, bool> predicate, bool destroy = true)
         /// <summary>
         /// Resets the <see cref="Player"/>'s stamina.
         /// </summary>
-        public void ResetStamina() => Stamina = StaminaStat.MaxValue;
+        /// <param name="multipliers">Reset <see cref="StaminaUsageMultiplier"/> and <see cref="StaminaRegenMultiplier"/> also.</param>
```
Comment (Misaka-ZeroTwo) [inline Exiled.API/Features/Player.cs:2031]:
```suggestion
        /// <param name="multipliers">Resets <see cref="StaminaUsageMultiplier"/> and <see cref="StaminaRegenMultiplier"/>.</param>
```
```suggestion
        /// <param name="multipliers">Resets <see cref="StaminaUsageMultiplier"/> and <see cref="StaminaRegenMultiplier"/>.</param>
```

**Pull request merged**